code/game/objects/structures/morgue.dm

```diff
@@ -223,6 +224,20 @@ GLOBAL_LIST_EMPTY(crematoriums)
 			update_icon()
 			playsound(src.loc, 'sound/machines/ding.ogg', 50, 1) //you horrible people
 
+/obj/structure/bodycontainer/crematorium/creamatorium
+	name = "creamatorium"
+	desc = "A human incinerator. Works well during ice cream socials."
+
+/obj/structure/bodycontainer/crematorium/creamatorium/cremate(mob/user)
+	var/list/icecreams = new()
+	for(var/mob/living/M in GetAllContents())
+		var/obj/item/weapon/reagent_containers/food/snacks/icecream/IC = new()
+		IC.set_cone_type("waffle")
+		IC.add_mob_flavor(M)
+		icecreams += IC
+	. = ..()
+	for(var/obj/IC in icecreams)
+		IC.loc = src

```

Comment:
forceMove, or you could just pass the loc in creation anyway ? Don't really see the need for another loop ?